src/Cache/MemoryCache.php

```diff
@@ -18,6 +18,15 @@ class MemoryCache extends Cache
 	 */
 	protected array $store = [];
 
+	/**
+	 * Returns whether the cache is ready to
+	 * store values beyond the current request
+	 */
+	public function enabled(): bool
+	{
+		return false;
```

Comment:
Initially I thought this new method is about right at this moment the cache is enabled... but this comment makes me think it's more about whether the cache will persist after the current request. In that case, I think the name is rather misleading. Maybe rather something like `isPersistent`.

Reply:
Good point.

The method currently only returns `true` if two conditions are fulfilled:

1. The cache is available at all.
2. The cache persists beyond the current request.

Maybe this combination is too confusing. What if we only make it check the first condition. The persistence part can then be checked in the calling methods by simply checking `instanceof NullCache` and `instanceof MemoryCache`.

Reply:
I have now implemented it so that the persistence no longer counts. This means that `MemoryCache` now returns `true`. The only cache that always returns false is `NullCache` as it's really never an active cache.

Reply:
I feel like `NullCache` should be `true` as well - it is designed to be used like an actual cache (just without doing anything). Might contradict situations where a cache type is unknown where checking for enabled would now fail, while so far the NullCache behaviour was the preferred way.

Reply:
Thinking about it again, I think returning `false` for `NullCache` was correct:

The only reason why we have this class is to return it as a placeholder when a cache is disabled via config. So any `NullCache` is by definition a disabled cache.

Reply:
I agree that NullCache should return false. It really is just a placeholder instead of not returning any cache instance on an inactive cache.

Reply:
@distantnative could you resolve the conversation if you agree?

Reply:
Agree